spring-data-jdbc/src/test/java/org/springframework/data/jdbc/repository/JdbcRepositoryIdGenerationIntegrationTests.java

```diff
@@ -29,31 +33,51 @@
 import org.springframework.context.annotation.FilterType;
 import org.springframework.context.annotation.Import;
 import org.springframework.data.annotation.Id;
+import org.springframework.data.annotation.PersistenceCreator;
+import org.springframework.data.annotation.Transient;
+import org.springframework.data.domain.Persistable;
 import org.springframework.data.jdbc.repository.config.EnableJdbcRepositories;
 import org.springframework.data.jdbc.repository.support.SimpleJdbcRepository;
+import org.springframework.data.jdbc.testing.AuditableNamedParameterJdbcTemplate;
 import org.springframework.data.jdbc.testing.IntegrationTest;
 import org.springframework.data.jdbc.testing.TestConfiguration;
 import org.springframework.data.relational.core.mapping.NamingStrategy;
+import org.springframework.data.relational.core.mapping.Sequence;
 import org.springframework.data.relational.core.mapping.event.BeforeConvertCallback;
 import org.springframework.data.repository.CrudRepository;
 import org.springframework.data.repository.ListCrudRepository;
 import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.test.context.jdbc.Sql;
+import org.testcontainers.shaded.org.checkerframework.checker.index.qual.SameLen;
 
 /**
  * Testing special cases for id generation with {@link SimpleJdbcRepository}.
  *
  * @author Jens Schauder
  * @author Greg Turnquist
+ * @author Mikhail Polivakha
  */
 @IntegrationTest
 class JdbcRepositoryIdGenerationIntegrationTests {
 
-	@Autowired NamedParameterJdbcTemplate template;
-	@Autowired ReadOnlyIdEntityRepository readOnlyIdRepository;
+    @Autowired ReadOnlyIdEntityRepository readOnlyIdRepository;
 	@Autowired PrimitiveIdEntityRepository primitiveIdRepository;
 	@Autowired ImmutableWithManualIdEntityRepository immutableWithManualIdEntityRepository;
 
-	@Test // DATAJDBC-98
+    @Autowired EntityWithSeqRepository entityWithSeqRepository;
+
+    @Autowired PersistableEntityWithSeqRepository persistableEntityWithSeqRepository;
+
+    @Autowired PrimitiveIdEntityWithSeqRepository primitiveIdEntityWithSeqRepository;
+
+    @Autowired AuditableNamedParameterJdbcTemplate auditableNamedParameterJdbcTemplate;
+
+    @BeforeEach
+    void setUp() {
+        auditableNamedParameterJdbcTemplate.clearRecordedStatements();
```

Comment:
When already having an integration test, it doesn't make much sense to verify against statements. Rather, we should the result.

Reply:
I'm sorry, but it does not seem to be clear for me how we can verify the `SELECT` statements executed to query the sequence.  We can create a wrapper around ID generating callback and try to utilize it. I am not sure which approach is better to be honest @mp911de

Reply:
There's a blog post by Vlad on verifying and counting statements see https://vladmihalcea.com/how-to-detect-the-n-plus-one-query-problem-during-testing/

I think that one can serve as input for decorating our data sources to be tested.